### PR TITLE
Full platform inference

### PR DESCRIPTION
**Pull request by marc0der**
- Align Platform inference with other projects
- Pre-emptive step to extracting Platform to domain library
- Lays the ground for Native Binary installation
- Some presentation tweaks on selfupdate

### CODE REVIEW

**Review by hgeraldino — APPROVED** [review]
lgtm

As a side question: are these new platform enums for the Mac M1? I recall having issues when running sdkman-cli's functional tests, where the UnixUtils was unable to match my architecture (uname -s) with these new values

**Comment by marc0der** [comment]
@hgeraldino the new `MacARM64` platform takes care of all of this on the serverside. We might need to revisit that piece of test code in the CLI as I don't think I've accommodated ARM64 Macs in the acceptance tests.

**Comment by marc0der** [comment]
@helpermethod you are right, although a lot of this code will be deleted soon anyway when we start using content headers. My plan is to remove all the Java hooks if possible. I'll fix the jmc ones though.

File: app/controllers/HooksController.scala [hunk 1]
```diff
@@ -37,42 +37,32 @@ class HooksController @Inject() (cc: ControllerComponents)
...
+          //POST: Linux 64 bit
+          case (Post, Java, _, LinuxX32 | LinuxX64 | LinuxARM64 | LinuxARM32HF | LinuxARM32SF, _) =>
+            Ok(views.txt.java_post_linux_tarball(candidate, version, platform))
+          case (Post, JMC, _, LinuxX32 | LinuxX64 | LinuxARM64 | LinuxARM32HF | LinuxARM32SF, _) =>
+            Ok(views.txt.jmc_post_unix_tarball(version, vendor, jmcBinaryExec(vendor, platform)))
+
+          //POST: Windows
+          case (Post, Java, "9", Windows64, OpenJDK) =>
+            Ok(views.txt.default_post_tarball(candidate, version, platform))
+          case (Post, Java, "10", Windows64, OpenJDK) =>
```
[the remark this answers is not in the document]
Comment (marc0der) [reply app/controllers/HooksController.scala:60]: This code will die soon so not going to spend too much time polishing it.

**Pull request merged**